samples/snippets/subscriber.py

```diff
@@ -540,9 +545,10 @@ def synchronous_pull_with_lease_management(project_id, subscription_id):
     subscriber = pubsub_v1.SubscriberClient()
     subscription_path = subscriber.subscription_path(project_id, subscription_id)
 
-    response = subscriber.pull(
-        request={"subscription": subscription_path, "max_messages": 3}
-    )
+    while not response:
```

Comment:
Perhaps have some limit so this doesn't take forever? Maybe use time, and just try for 5 min max?

Reply:
I set the retry deadline instead of using a while loop.